src/signature.rs

```diff
@@ -76,24 +79,17 @@ fn check_scalar(bytes: [u8; 32]) -> Result<Scalar, SignatureError> {
         return Err(InternalError::ScalarFormat.into());
     }
 
+    // You cannot do arithmetic with scalars construct with Scalar::from_bits. We only use this
+    // scalar for EdwardsPoint::vartime_double_scalar_mul_basepoint, which is an accepted usecase.
+    // The `from_bits` method is deprecated because it's unsafe. We know this.
+    #[allow(deprecated)]
     Ok(Scalar::from_bits(bytes))
 }
 
+/// Ensures that the scalar `s` of a signature is within the bounds [0, ℓ)
 #[cfg(not(feature = "legacy_compatibility"))]
```

Comment:
This might be a little easier to read if it were written with an `if` statement instead of having two definitions depending on the feature. Something like:

```rust
fn check_scalar(bytes: [u8; 32]) -> Result<Scalar, SignatureError> {
    if cfg!(feature = "legacy_compatibility") {
        [...]
    } else {
        [...]
     }
}
```

Reply:
I played with this a bit, and, because of the extensive comments, I thought it looked better the way it is currently. Is it ok if we keep it?

Reply:
Sure